trash.go

```diff
@@ -128,12 +130,75 @@ func run(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
+
 	if update {
 		return updateTrash(trashDir, dir, targetDir, confFile, trashConf, insecure)
 	}
+
 	if err := vendor(keep, trashDir, dir, targetDir, trashConf, insecure); err != nil {
 		return err
 	}
+
+	var extraImports []conf.Import
+	for _, packageImport := range trashConf.Imports {
+		if packageImport.Transitive {
+			repoDir := path.Join(trashDir, "src", packageImport.Package)
+			transitiveDependencies, err := godep.Parse(repoDir)
+			if err != nil {
+				return err
+			}
+			for _, transitiveDependency := range transitiveDependencies {
+				extraImports = append(extraImports, conf.Import{
+					Package: transitiveDependency.Name,
+					Version: transitiveDependency.Reference,
+					Repo:    transitiveDependency.Repository,
+				})
+			}
+		}
+	}
+
+	var filteredExtraImports []conf.Import
+	for _, extraImport := range extraImports {
+		packageAlreadyImported := false
+		for _, packageImport := range trashConf.Imports {
+			if packageImport.Package == extraImport.Package {
+				packageAlreadyImported = true
+				break
+			}
+		}
+		if !packageAlreadyImported {
+			filteredExtraImports = append(filteredExtraImports, extraImport)
+		}
+	}
+	trashConf.Imports = append(trashConf.Imports, filteredExtraImports...)
+
+	if err := vendor(keep, trashDir, dir, targetDir, trashConf, insecure); err != nil {
+		return err
+	}
+
+	vendorDir := path.Join(dir, targetDir)
+	for _, packageImport := range trashConf.Imports {
+		if !packageImport.Staging {
+			continue
+		}
+
+		packageLocation := path.Dir(packageImport.Package)
+		baseDir := path.Join(trashDir, "src", packageImport.Package, "staging/src", packageLocation)
+
+		files, err := ioutil.ReadDir(baseDir)
+		if err != nil {
+			return err
```

Comment:
Good question. This block is taken from [here](https://github.com/joshwget/trash/blob/f74cda7e1218923ce8dc475d0dcb8cccc0834571/trash.go#L399), and I think we should really be returning an error in all cases.